src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const rules = sourceString.split(';');
+
+  rules.forEach((rule) => {
+    const clearRule = rule.trim();
+
+    if (clearRule) {
+      const parts = clearRule.split(':');
+      const key = parts[0].trim();
+      const value = parts.length > 1 ? parts.slice(1).join(':').trim() : '';
+
+      if (key) {
+        result[key] = value;
```

Comment:
Ensure that the key is not an empty string before assigning it to the result object. The current check only ensures that `key` is truthy, which might not be sufficient if the input string contains keys that are just whitespace.